### ISSUE TITLE
Application support: can't add the second student

### ISSUE THREAD

**Issue by mkalininait** [rails-girls-summer-of-code/rgsoc-teams#434]
One student writes in Slack:
"Hi, I'm trying to add another student to my team, but get an error
https://teams.railsgirlssummerofcode.org/teams/121
1 error prohibited this team from being saved:
Roles there cannot be more than 2 students on a team.
Although there is currently only one student (me)"

I looked at it, there is one blank student (without a github handle), and I can’t neither add a github handle, nor delete it from the team.

**Comment by alicetragedy** [comment]
I tried it locally again, and was able to add a member to my team without github handle. we should probably fix this. thanks for noticing @mkalininait!

**Comment by alicetragedy** [comment]
Update: since validations are in place to ensure that the github handle isn't blank, this wasn't the problem. A user is required for the role (so, to be added to the team) and if the user is not being present in the database it silently fails (and adds an empty role to the team). 
Not sure what the fix is, (or if there is one) but we will make extra sure that students know they all have to log in in order to be added to a team.

**Comment by sainera** [comment]
My team was trying to submit the application the whole day but even after completing the application process, it was getting saved as the draft, By the time we tried looking for the issue and retry he application process,the application got closed. Please tell me how to submit the application as it got closed now.
Please help.

**Comment by alicetragedy** [comment]
@sainera I'm sorry, but this is not the right place for application support. please get in touch with us by email. You can find the contact information on our [about page](http://railsgirlssummerofcode.org/about/).

**Comment by mkalininait** [comment]
Tested on Staging: still can add a role without a github handle. 
"...and I can’t neither add a github handle, nor delete it from the team." <- I still can't add a handle, but I can delete this empty role at least.

**Comment by alicetragedy** [comment]
just a thought: If you are testing this on staging (where you might have a “student” role as well as an “organiser” role) that might be the reason why you can delete the empty role and a “simple” student might not be able to. have you tried testing locally with a “login as user” feature (or changing your role to student-only)?

**Comment by mkalininait** [comment]
@alicetragedy I have a special user for testing -- mkalininait-test. It doesn't have any special permissions.

**Comment by alicetragedy** [comment]
I'm sorry @mkalininait, I had no idea about that. then it looks like it was magically half-fixed at least 😂 (I think that not being able to remove the role was the more critical/annoying part of the bug, so it might not be an issue that needs an urgent fix after all if students can remove the empty roles themselves). Thanks for testing it again!

**Comment by F3PiX** [comment]
Yeah, removing a role was a separate issue  and was solved in # 577

**Comment by carpodaster** [comment]
Can this be closed?

**Comment by mkalininait** [comment]
@carpodaster 
We still can add a role without a github handle. It's not a huge deal, and we could live with this.
But I'm not sure who on the RGSoC orga team makes (is eligible to make) decisions about product features and bugs.

**Comment by LuisaAPF** [comment]
I was not able to reproduce this issue. If I try to assign a role to an empty github handle, I get a 500 error and no role is created on the database. I can work instead on fixing the validations.

**Comment by carpodaster** [comment]
@LuisaAPF thank you for offering to fix this issue! There is an open PR (#796) that could be related. Let's wait until it is merged and see if the problem persists.

**Comment by LuisaAPF** [comment]
Ok!

**Comment by F3PiX** [comment]
@LuisaAPF I happen to notice that you picked one of the few issues that are labeled `beginner friendly`, and not `rgsoc17`. You are very welcome to pick issues with the `rgsoc17` label as well. As far as I know, they are not exclusive for the RGSoC team. Am I right, @carpodaster ?

**Comment by carpodaster** [comment]
>  As far as I know, they are not exclusive for the RGSoC team. Am I right, @carpodaster ?

@F3PiX @LuisaAPF definitely – everything that's not been assigned (or being commented on that someone's working on it) is up for grabs

**Comment by LuisaAPF** [comment]
Thanks for the tip, @F3PiX. I wasn't sure I could work on those.

**Comment by F3PiX** [comment]
@LuisaAPF If you have questions, want to check for suitable issues, or could use some help with the setup or anything else that may hold you back, please don't hesitate to ask!! You can put a question in an issue, and someone will answer, and/or 'ping' me ( i.e. add ` @F3PiX` to your comment).
